many/src/List.js

```diff
@@ -1,100 +1,58 @@
 import xs from 'xstream';
 import {button, div} from '@cycle/dom';
-import isolate from '@cycle/isolate'
+import Collection from '@cycle/collection';
 import Item from './Item';
 
-function intent(DOM, itemRemove$) {
-  return xs.merge(
-    DOM.select('.add-one-btn').events('click')
-      .mapTo({type: 'ADD_ITEM', payload: 1}),
-
-    DOM.select('.add-many-btn').events('click')
-      .mapTo({type: 'ADD_ITEM', payload: 1000}),
-
-    itemRemove$.map(id => ({type: 'REMOVE_ITEM', payload: id}))
-  );
-}
-
-function model(action$, itemFn) {
-  function createRandomItemProps() {
-    let hexColor = Math.floor(Math.random() * 16777215).toString(16);
-    while (hexColor.length < 6) {
-      hexColor = '0' + hexColor;
-    }
-    hexColor = '#' + hexColor;
-    const randomWidth = Math.floor(Math.random() * 800 + 200);
-    return {color: hexColor, width: randomWidth};
+function createRandomItemProps() {
+  let hexColor = Math.floor(Math.random() * 16777215).toString(16);
+  while (hexColor.length < 6) {
+    hexColor = '0' + hexColor;
   }
+  hexColor = '#' + hexColor;
+  const randomWidth = Math.floor(Math.random() * 800 + 200);
+  return {Props: xs.of({color: hexColor, width: randomWidth})};
+}
 
-  let mutableLastId = 0;
+function makeItems(count) {
+  const items = [];
 
-  function createNewItem(props) {
-    const id = mutableLastId++;
-    const sinks = itemFn(props, id);
-    return {id, DOM: sinks.DOM.remember(), Remove: sinks.Remove};
+  for (let i = 0; i < count; i += 1) {
+    items.push(createRandomItemProps());
   }
 
-  const addItemReducer$ = action$
-    .filter(a => a.type === 'ADD_ITEM')
-    .map(action => {
-      const amount = action.payload;
-      let newItems = [];
-      for (let i = 0; i < amount; i++) {
-        newItems.push(createNewItem(createRandomItemProps()));
-      }
-      return function addItemReducer(listItems) {
-        return listItems.concat(newItems);
-      };
-    });
-
-  const removeItemReducer$ = action$
-    .filter(a => a.type === 'REMOVE_ITEM')
-    .map(action => function removeItemReducer(listItems) {
-      return listItems.filter(item => item.id !== action.payload);
-    });
+  return items;
+}
 
-  const initialState = [createNewItem({color: 'red', width: 300})]
+function intent(DOM) {
+  return xs.merge(
+    DOM.select('.add-one-btn').events('click')
+      .map(() => makeItems(1)),
 
-  return xs.merge(addItemReducer$, removeItemReducer$)
-    .fold((listItems, reducer) => reducer(listItems), initialState);
+    DOM.select('.add-many-btn').events('click')
+      .map(() => makeItems(1000))
+  );
 }
 
 function view(items$) {
-  const addButtons = div('.addButtons', [
-    button('.add-one-btn', 'Add New Item'),
-    button('.add-many-btn', 'Add Many Items')
-  ]);
-
-  return items$.map(items => {
-    const itemVNodeStreamsByKey = items.map(item =>
-      item.DOM.map(vnode => {
-        vnode.key = item.id; return vnode;
-      })
+  return Collection.pluck(items$, 'DOM')
```

Comment:
Say we have 10 items, and the third is removed. If the vnodes aren't keyed, this may lead to the suboptimal update of vnodes from 3 to 9, instead of just removing the third.